packages/react-api/src/api/model.js

```diff
@@ -66,15 +66,46 @@ export function executeModel(props) {
     filtersLogicalOperator
   };
 
-  // API supports multiple filters, we apply it only to geoColumn
-  const spatialFilters = spatialFilter
-    ? {
-        [source.geoColumn ? source.geoColumn : DEFAULT_GEO_COLUMN]: spatialFilter
+  let spatialFilters;
+  if (spatialFilter) {
+    let spatialDataType = source.spatialDataType;
+    let spatialDataColumn = source.spatialDataColumn;
+
+    if (!spatialDataType || !spatialDataColumn) {
+      if (source.geoColumn) {
+        const parsedGeoColumn = source.geoColumn ? source.geoColumn.split(':') : [];
+        if (parsedGeoColumn.length === 2) {
+          spatialDataType = parsedGeoColumn[0];
+          spatialDataColumn = parsedGeoColumn[1];
+        } else if (parsedGeoColumn.length === 1) {
+          spatialDataColumn = parsedGeoColumn[0] || DEFAULT_GEO_COLUMN;
+          spatialDataType = 'geo';
+        }
```

Comment:
Could `parsedGeoColumn` really be `[]`? If so do we need a third case for it?

Reply:
I don't think so (and i've checked as i wasn't sure):

```
> "".split(':')
['']
> "a".split(':')
['a']
> "a:b".split(':')
['a', 'b']
```